homeassistant/components/comelit/humidifier.py

```diff
@@ -0,0 +1,219 @@
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit humidifiers."""
+
+    coordinator: ComelitSerialBridge = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[ComelitHumidifierEntity] = []
+    for device_class in (
+        HumidifierDeviceClass.HUMIDIFIER,
+        HumidifierDeviceClass.DEHUMIDIFIER,
+    ):
+        entities.extend(
+            ComelitHumidifierEntity(
+                coordinator, device, config_entry.entry_id, device_class.value
+            )
+            for device in coordinator.data[CLIMATE].values()
+        )
+
+    async_add_entities(entities)
+
+
+class ComelitHumidifierEntity(CoordinatorEntity[ComelitSerialBridge], HumidifierEntity):
+    """Humidifier device."""
+
+    _attr_supported_features = HumidifierEntityFeature.MODES
+    _attr_available_modes = [MODE_NORMAL, MODE_AUTO]
+    _attr_min_humidity = 10
+    _attr_max_humidity = 90
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: ComelitSerialBridge,
+        device: ComelitSerialBridgeObject,
+        config_entry_entry_id: str,
+        device_class: str,
+    ) -> None:
+        """Init light entity."""
+        self._api = coordinator.api
+        self._device = device
+        super().__init__(coordinator)
+        # Use config_entry.entry_id as base for unique_id
+        # because no serial number or mac is available
+        self._attr_unique_id = f"{config_entry_entry_id}-{device.index}-{device_class}"
+        self._attr_device_info = coordinator.platform_device_info(device, device_class)
+        if device_class == HumidifierDeviceClass.DEHUMIDIFIER.value:
+            self._attr_device_class = HumidifierDeviceClass.DEHUMIDIFIER
+            self._attr_translation_key = "dehumidifier"
+        else:
+            self._attr_device_class = HumidifierDeviceClass.HUMIDIFIER
+            self._attr_translation_key = "humidifier"
+
+    @property
+    def _humidifier(self) -> list[Any]:
+        """Return humidifier device data."""
+        # CLIMATE has a 2 item tuple:
+        # - first  for Clima
+        # - second for Humidifier
+        return self.coordinator.data[CLIMATE][self._device.index].val[1]
+
+    @property
+    def _api_mode(self) -> str:
+        """Return device mode."""
+        # Values from API: "O", "L", "U"
+        return self._humidifier[2]
+
+    @property
+    def _api_active(self) -> bool:
+        "Return device active/idle."
+        return self._humidifier[1]
+
+    @property
+    def _api_automatic(self) -> bool:
+        """Return device in automatic/manual mode."""
+        return self._humidifier[3] == HumidifierComelitMode.AUTO
+
+    @property
+    def _is_dehumidifier(self) -> bool:
+        """Return true if device is set as dehumidifier."""
+        return (
+            self._api_mode == HumidifierComelitMode.LOWER
+            and self._attr_device_class == HumidifierDeviceClass.DEHUMIDIFIER
```

Comment:
I would pass in the active api mode in the class constructor. So you need less branches in the code.